lib/godmin/resources/resource_service.rb

```diff
@@ -17,14 +17,13 @@ module ResourceService
 
       attr_reader :options
 
-      def initialize(resource_class = nil, options = {})
+      def initialize(options = {})
         @options = options
-        @resource_class = resource_class
       end
 
       # TODO: should this raise its own error?
       def resource_class
-        @resource_class || self.class.name.demodulize.chomp("Service").constantize
+        @options[:resource_class] || self.class.name.demodulize.chomp("Service").constantize
```

Comment:
Ah, you just moved it to the options hash in case someone wants to pass it in? When doing something custom and fancy?